src/Lykke.Service.Zcash.Api.AzureRepositories/Addresses/AddressRepository.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AzureStorage;
+using AzureStorage.Tables;
+using Common.Log;
+using Lykke.Service.Zcash.Api.Core.Domain;
+using Lykke.Service.Zcash.Api.Core.Domain.Addresses;
+using Lykke.SettingsReader;
+
+namespace Lykke.Service.Zcash.Api.AzureRepositories.Addresses
+{
+    public class AddressRepository : IAddressRepository
+    {
+        private INoSQLTableStorage<AddressEntity> _tableStorage;
+        private static string GetPartitionKey(ObservationCategory category) => Enum.GetName(typeof(ObservationCategory), category);
+        private static string GetRowKey(string address) => address;
+
+        public AddressRepository(IReloadingManager<string> connectionStringManager, ILog log)
+        {
+            _tableStorage = AzureTableStorage<AddressEntity>.Create(connectionStringManager, "ZcashObservableAddresses", log);
+        }
+
+        public async Task CreateAsync(ObservationCategory category, string address)
+        {
+            var partitionKey = GetPartitionKey(category);
+            var rowKey = GetRowKey(address);
+
+            await _tableStorage.InsertAsync(new AddressEntity(partitionKey, rowKey));
+        }
+
+        public async Task DeleteAsync(ObservationCategory category, string address)
```

Comment:
Fixed
But DeleteIfExistsAsync always return true, even if record not exists. As a result current implementation violates blockchain API contract - it returns 200 instead of 204 if record was deleted earlier.